wagtail/wagtailcore/blocks/base.py

```diff
@@ -204,24 +207,100 @@ def get_prep_value(self, value):
         return value
 
     def get_context(self, value):
+        """
+        Return a dict of context variables (derived from the block value, or otherwise)
+        to be added to the template context when rendering this value through a template.
+        """
         return {
             'self': value,
             self.TEMPLATE_VAR: value,
         }
 
-    def render(self, value):
+    def _render_with_context(self, value, context=None):
+        """
+        Temporary hack to accommodate Block subclasses created for Wagtail <1.6 which
+        have overridden `render` with the type signature:
+            def render(self, value)
+        and will therefore fail when passed a `context` kwarg.
+
+        In Wagtail 1.8, when support for context-less `render` methods is dropped,
+        this method will be deleted (and calls to it replaced with a direct call to `render`).
+        """
+        try:
+            return self.render(value, context=context)
+        except TypeError:
```

Comment:
> just to handle this very obscure special case.

It may be just typo or "4-rd party code" can rise`TypeError`... For example, I've tried to [add a pagination](https://github.com/torchbox/wagtail/pull/2786#issuecomment-230416360) to a block. It's a bad solution because I pass request object into my block through templates, but it's a good example when it will be hard to figure out what's going on.

```python
    def render(self, value, context=None):
        request = context.get('request') if context else None
        if not request:
            raise Exception('Sad trombone sound')

        pages = context.get('pages')

        per_page = settings.DEFAULT_PER_PAGE
        paginator = Paginator(pages, per_page)
        page_number = request.GET.get('page')

        try:
            pages = paginator.page(page_number)
        except PageNotAnInteger:
            # Hint: TypeError will be raised somewhere in Django's Paginator
            pages = paginator.page(1)
        except EmptyPage:
            pages = paginator.page(paginator.num_pages)

        context = dict(context)
        context['pages'] = pages

        return super().render(value, context)
```